AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreConfiguration.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+/// Error Handler function typealias
+public typealias DataStoreErrorHandler = (AmplifyError) -> Void
+
+/// Holds a reference to both the local `Model` and the remote one during a conflict
+/// resolution. Implementations of the `DataStoreConflictHandler` use this to decide
+/// what the outcome of a conflict should be.
+public struct DataStoreConclictData {
+    public let local: Model
+    public let remote: Model
+}
+
+/// Conflict Handler function typealias. The function is used during a conflict that
+/// could not be resolved and requires a decision from the consumer.
+public typealias DataStoreConflictHandler = (DataStoreConclictData, DataStoreConflictHandlerResolver) -> Void
+
+/// Callback for the `DataStoreConflictHandler`.
+public typealias DataStoreConflictHandlerResolver = (DataStoreConflictHandlerResult) -> Void
+
+/// The conflict resolution result enum.
+public enum DataStoreConflictHandlerResult {
+
+    /// Discard the local changes in favor of the remote ones.
+    case discard
```

Comment:
We should be very explicit: If this discards local changes in favor of remote ones, does that mean updating the system, at the time of the conflict resolution handler, with the state of the remote item? If so then let's document that. Better yet, maybe more appropriate names would be: `applyRemote`, `retryLocal`, and `retry(Model)`?